plugins/serializers/wavefront/wavefront.go

```diff
@@ -0,0 +1,196 @@
+package wavefront
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs/wavefront"
+)
+
+// WavefrontSerializer : WavefrontSerializer struct
+type WavefrontSerializer struct {
+	Prefix         string
+	UseStrict      bool
+	SourceOverride []string
+}
+
+// catch many of the invalid chars that could appear in a metric or tag name
+var sanitizedChars = strings.NewReplacer(
+	"!", "-", "@", "-", "#", "-", "$", "-", "%", "-", "^", "-", "&", "-",
+	"*", "-", "(", "-", ")", "-", "+", "-", "`", "-", "'", "-", "\"", "-",
+	"[", "-", "]", "-", "{", "-", "}", "-", ":", "-", ";", "-", "<", "-",
+	">", "-", ",", "-", "?", "-", "/", "-", "\\", "-", "|", "-", " ", "-",
+	"=", "-",
+)
+
+// catch many of the invalid chars that could appear in a metric or tag name
+var strictSanitizedChars = strings.NewReplacer(
+	"!", "-", "@", "-", "#", "-", "$", "-", "%", "-", "^", "-", "&", "-",
+	"*", "-", "(", "-", ")", "-", "+", "-", "`", "-", "'", "-", "\"", "-",
+	"[", "-", "]", "-", "{", "-", "}", "-", ":", "-", ";", "-", "<", "-",
+	">", "-", "?", "-", "\\", "-", "|", "-", " ", "-", "=", "-",
+)
+
+var tagValueReplacer = strings.NewReplacer("\"", "\\\"", "*", "-")
+
+var pathReplacer = strings.NewReplacer("_", ".")
+
+func NewSerializer(prefix string, useStrict bool, sourceOverride []string) (*WavefrontSerializer, error) {
+	s := &WavefrontSerializer{
+		Prefix:         prefix,
+		UseStrict:      useStrict,
+		SourceOverride: sourceOverride,
+	}
+	return s, nil
+}
+
+// Serialize : Serialize based on Wavefront format
+func (s *WavefrontSerializer) Serialize(m telegraf.Metric) ([]byte, error) {
+	out := []byte{}
+	metricSeparator := "."
+
+	for fieldName, value := range m.Fields() {
+		var name string
+
+		if fieldName == "value" {
+			name = fmt.Sprintf("%s%s", s.Prefix, m.Name())
+		} else {
+			name = fmt.Sprintf("%s%s%s%s", s.Prefix, m.Name(), metricSeparator, fieldName)
+		}
+
+		if s.UseStrict {
+			name = strictSanitizedChars.Replace(name)
+		} else {
+			name = sanitizedChars.Replace(name)
+		}
+
+		name = pathReplacer.Replace(name)
+
+		metric := &wavefront.MetricPoint{
+			Metric:    name,
+			Timestamp: m.Time().Unix(),
+		}
+
+		metricValue, buildError := buildValue(value, metric.Metric)
+		if buildError != nil {
+			log.Printf("E! Serializer [wavefront] %s\n", buildError.Error())
```

Comment:
I think this should be a debug level message, or possibly silent at least for string fields.
```
D! [serializers.wavefront] Serialization error: %v
```

Reply:
I moved log output to buildValue and only do a debug log when the default case is triggered.  Strings will silently error but continue to next metric.